Add example to environment docs (#2420)
jenkins-infra/jenkins.io: content/doc/book/pipeline/jenkinsfile.adoc

assuming a Jenkins master is running on `localhost:8080`, and includes:

BUILD_ID:: The current build ID, identical to BUILD_NUMBER for builds created in Jenkins versions 1.597+
BUILD_NUMBER:: The current build number, such as "153"

> BUILD_TAG:: String of jenkins-\$\{JOB_NAME}-\$\{BUILD_NUMBER}. Convenient to put into a resource file, a jar file, etc for easier identification

Author: No need to escape both.  When both are escaped, the backslash renders on the page.
```suggestion
BUILD_TAG:: String of jenkins-$\{JOB_NAME}-$\{BUILD_NUMBER}. Convenient to put into a resource file, a jar file, etc for easier identification
```

BUILD_URL:: The URL where the results of this build can be found (for example \http://buildserver/jenkins/job/MyJobName/17/ )
EXECUTOR_NUMBER:: The unique number that identifies the current executor (among executors of the same machine) performing this build. This is the number you see in the "build executor status", except that the number starts from 0, not 1
JAVA_HOME:: If your job is configured to use a specific JDK, this variable is set to the JAVA_HOME of the specified JDK. When this variable is set, PATH is also updated to include the bin subdirectory of JAVA_HOME

> JENKINS_URL:: Full URL of Jenkins, such as https://example.com:port/jenkins/ (NOTE: only available if Jenkins URL set in "System Configuration")

Author: Don't make example.com references into clickable URL's.  They are designed to not go anywhere.
```suggestion
JENKINS_URL:: Full URL of Jenkins, such as \https://example.com:port/jenkins/ (NOTE: only available if Jenkins URL set in "System Configuration")
```

given environment variables to steps within the `stage`.

==== Setting environment variables dynamically

> Environment variables can be set at run time with shell scripts (`sh`), Windows batch scripts (`bat`) and PowerShell scripts (`powershell`).

Author: Environment variables in this context are set by the Pipeline and *used by* the scripts.  As far as I know, there isn't an obvious way for a script to set an environment variable and have it persist in the Pipeline for use in later scripts.
```suggestion
Environment variables can be set at run time and can be used by shell scripts (`sh`), Windows batch scripts (`bat`) and PowerShell scripts (`powershell`).
```
